include/systems/equation_systems.h

```diff
@@ -454,6 +454,24 @@ class EquationSystems : public ReferenceCountedObject<EquationSystems>,
    **/
   void allgather ();
 
+  /**
+   * Calls to reinit() will also do two-step coarsen-then-refine
+   **/
+  void enable_refine_in_reinit()
```

Comment:
@roystgnr and @jwpeterson will have the final word, but I would've preferred just a traditional `set/get_refine_in_reinit_flag()` accessor pattern myself.